addons/base_automation/models/base_automation.py

```diff
@@ -30,25 +30,97 @@
     False: 0,
 }
 
+CREATE_TRIGGERS = [
+    'on_create',
+
+    'on_create_or_write',
+    'on_priority_set',
+    'on_stage_set',
+    'on_state_set',
+    'on_tag_set',
+    'on_user_set',
+]
+
+WRITE_TRIGGERS = [
+    'on_write',
+    'on_archive',
+    'on_unarchive',
+
+    'on_create_or_write',
+    'on_priority_set',
+    'on_stage_set',
+    'on_state_set',
+    'on_tag_set',
+    'on_user_set',
+]
+
+TIME_TRIGGERS = [
+    'on_time',
+    'on_time_created',
+    'on_time_updated',
+]
```

Comment:
Unfortunately we also use these globals in domains and the ORM doesn't support sets in domains.
```py
automations = self.with_context(active_test=True).search([('trigger', 'in', TIME_TRIGGERS)])
```
We could cast those sets in list where we use it in domains, but It would defeat the purpose of using sets